trin-core/src/portalnet/storage.rs

```diff
@@ -0,0 +1,605 @@
+        ).unwrap();
+
+    }
+
+    fn meta_db_remove(&self, content_id: &[u8; 32]) {
+
+        self.meta_db.execute(
+            DELETE_QUERY,
+            [content_id.to_vec()],
+        ).unwrap();
+        
+    }
+
+    pub fn store(&mut self, key: &String, value: &String) {
+
+        // Check whether we should store this data
+        if !self.should_store(&key) {
+            println!("Not storing: {}", key);
+            return;
+        }   
+
+        let content_id = self.content_key_to_content_id(key);
+
+        // Store the data 
+        self.db.put(&content_id, value).expect("Failed to write to DB");
+        self.meta_db_insert(&content_id, &key, value);
+
+        // Update the farthest key if this key is either 1.) the first key ever or 2.) farther than the current farthest
+        match self.farthest_content_id.as_ref() {
+            None => {
+                self.farthest_content_id = Some(content_id);
+            },
+            Some(farthest) => {
+                if self.distance_to_content_id(&content_id) > self.distance_to_content_id(&farthest) {
+                    self.farthest_content_id = Some(content_id.clone());
+                }
+            }
+        }
+
+        // Delete furthest data until our data usage is less than capacity
+        while self.capacity_reached() {
+
+            let id_to_remove = self.farthest_content_id.clone();
+            
+            println!("Capacity reached, deleting farthest: {}", hex::encode(&id_to_remove.clone().unwrap()));
+
+            self.db.delete(&id_to_remove.clone().unwrap()).expect("Failed to delete key.");
+            self.meta_db_remove(&id_to_remove.clone().unwrap());
+
+            self.db.flush().expect("Failed to flush db.");
+            
+            match self.find_farthest_content_id() {
+                None => { error!("Failed to find farthest!") },
+                Some(farthest) => {
+                    println!("Found new farthest: {}", hex::encode(&farthest));
+                    self.farthest_content_id = Some(farthest.clone());
+                    self.data_radius = self.distance_to_content_id(&farthest);
+                }
+            }
+
+        } 
+
+    }
+
+    pub fn get(&self, key: &String) -> Result<Option<Vec<u8>>, Error> {
+
+        let content_id = self.content_key_to_content_id(key);
+        self.db.get(content_id)
+
+    }
+
+    pub fn get_current_radius(&self) -> u64 {
+
+        self.data_radius
+
+    }
+
+    pub fn capacity_reached(&self) -> bool {
+
+        let storage_usage = self.get_total_storage_usage_in_bytes_from_network().unwrap();
+        storage_usage > (self.storage_capacity_kb * 1000)
+
+    }
+
+    pub fn get_total_storage_usage_in_bytes_on_disk(&self) -> u64 {
+
+        self.get_total_size_of_directory_in_bytes(get_data_dir()).unwrap()
+
+    }
+
+    pub fn get_total_storage_usage_in_bytes_from_network(&self) -> Result<u64, String> {
```

Comment:
Since this is returning a `Result` anyway, we might as well bubble up the errors instead of `unwrap` to panic on them.  (I'm thinking of the `?` operator here)